internal/pkg/bootstrap/handlers/secret.go

```diff
@@ -16,42 +16,50 @@ package handlers
 
 import (
 	"context"
+	"github.com/edgexfoundry/go-mod-secrets/pkg"
 	"os"
 	"sync"
 
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
-	"github.com/edgexfoundry/go-mod-registry/registry"
-	"github.com/edgexfoundry/go-mod-secrets/pkg"
-	"github.com/edgexfoundry/go-mod-secrets/pkg/providers/vault"
-
-	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/container"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/startup"
-)
+	"github.com/edgexfoundry/edgex-go/internal/pkg/di"
 
-// Environment variable used to disable the secret store functionality which is ON/TRUE by default.
-const SecretStore = "EDGEX_SECURITY_SECRET_STORE"
-
-// SecretClient is a global variable used to house the SecretClient to be used system-wide for obtaining secrets.
-var SecretClient pkg.SecretClient
+	"github.com/edgexfoundry/go-mod-secrets/pkg/providers/vault"
+)
 
-// SecretClientBootstrapHandler creates a SecretClient to be used for obtaining secrets from a secret store manager.
+// SecretClientBootstrapHandler creates a secretClient to be used for obtaining secrets from a secret store manager.
 // NOTE: This BootstrapHandler is responsible for creating a utility that will most likely be used by other
 // BootstrapHandlers to obtain sensitive data, such as database credentials. This BootstrapHandler should be processed
 // before other BootstrapHandlers, possibly even first since it has not other dependencies.
 func SecretClientBootstrapHandler(
 	wg *sync.WaitGroup,
 	context context.Context,
 	startupTimer startup.Timer,
-	config interfaces.Configuration,
-	loggingClient logger.LoggingClient,
-	registryClient registry.Client) bool {
+	dic *di.Container) bool {
 
-	if env := os.Getenv(SecretStore); env != "false" {
+	// check for environment variable that turns security off
+	if env := os.Getenv("EDGEX_SECURITY_SECRET_STORE"); env != "false" {
```

Comment:
`EDGEX_SECURITY_SECRET_STORE` should be a constant. Note that docker-edgex-mongo also uses this.

https://github.com/edgexfoundry/docker-edgex-mongo/blob/0769d9f0c8416f69ea421b9bf45ba0e6c6735a44/internal/pkg/constants.go#L20

I think that means defining it in `go-mod-secrets` would make the most sense. @AnthonyMBonafide for visibility.

Reply:
Agree but also believe this should be worked separately.  Created https://github.com/edgexfoundry/go-mod-secrets/issues/32